runtime/generator.go

```diff
@@ -44,6 +44,7 @@ type Generator struct {
 
 // NewGenerator returns a new Generator object that runs the given Block b.
 func NewGenerator(f *Frame, fn func(*Object) (*Object, *BaseException)) *Generator {
+	f.taken = true // Claim the frame from being returned.
```

Comment:
There are a few other cases where taken needs to be set:

* frame.f_back access
* `__frame__()` builtin
* newTraceback()